File: codechef/February Cookoff 2k18/3.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int t, n, k, b;
    cin >> t;
    while (t--)
    {
        cin >> n >> k >> b;
        int a[n];
        for (int i = 0; i < n; i++)
            cin >> a[i];
        sort(a, a + n);
        int curr = 0;
        int next = 1;
        int count = 1;
        while (next < n)
        {
            if (a[curr] * k + b <= a[next])
            {
                curr++;
                next++;
                count++;
            }
            else
                next++;
        }
        cout << count << endl;
    }
    return 0;
}

File: codechef/February Cookoff 2k18/3.py
t = int(raw_input())

for _ in xrange(t):
    n, k, b = map(int, raw_input().split())
    a = map(int, raw_input().split())
    a.sort()
    curr = 0
    next = 1
    count = 1
    while next < n:
        if a[curr] * k + b <= a[next]:
            count += 1
            next += 1
            curr += 1
        else:
            next += 1
    print count
